### DISCUSSION TITLE
Disable "Remember opened files" but keep access to Settings

### DISCUSSION THREAD

**Discussion by GarethThom** [sumatrapdfreader/sumatrapdf#3025] [Q&A]
Hi,

We would like the default for the "Remember opened files" setting to be "No", but the only way we can find of achieving this is to specify the "SavePreferences = 0" setting in the restricted use file (sumatrapdfrestrict.ini). However, we still want people to be able to change their preferred Default Layout and Default Zoom, for example, which this also restricts.

This is for running in a terminal server environment where hundreds of invoice PDFs are opened each day (for allocation back to Purchase Orders/GRNs) have having it remember previously opened files will start using up lots of RAM.

Is there a solution/workaround that allows this?

Let me know, thanks,
Gareth

**Comment by GitHubRulesOK** [comment]
without restrictions you can set it to no in options or settings, but restrictions are naturally enblock to ensure the block of related features are not bypassed.
![image](https://user-images.githubusercontent.com/12754913/196184047-6a901cf2-d8e0-4841-afe1-25064ea01b36.png)

**Reply by GarethThom** [reply]
Yes - you can manually untick the "Remember opened files" flag, but the point is that the settings file is stored per user and is on by default, meaning that all existing terminal server users as well as any new users in the future would have to go in an explicitly turn this off. I was looking for a way to essentially default this to off for all users.

**Reply by GitHubRulesOK** [reply]
you can pre-seed sets of settings during install (you must be if adding restrict ini) even that one line on its own may be respected if you match utf-8 BOM

```
RememberOpenedFiles = false
RememberStatePerDocument = true
```
on opening those will be remembered and later added to on close
![image](https://user-images.githubusercontent.com/12754913/196205962-f37754e2-f0ae-48d4-af3b-57ab65cc7396.png)

**Reply by GarethThom** [reply]
Thanks for getting back to me. Our plan is to install using the "-all-users" command line argument so it is only installed once on the terminal server and available for all users (current and future), so we do not have any kind of pre-seeded settings file at all at the moment.

In order to achieve that we would need to add some kind of Group Policy setting or login script that creates the "%USERPROFILE%\AppData\Local\SumatraPDF\SumatraPDF-settings.txt" file with the relevant seed values.

Is this the only way of handling this or can you suggest anything else?

Thanks!

**Reply by GitHubRulesOK** [reply]
I don't run central installs my methods are served by batchfiles to pull downloads and distribute between user locations

for a domain you must be using login scripting where a one time find and replace then self destroy repeat command would be possible, avoid edit existing settings (unless using file editor) as modify BOM often causes problems
Start with ALL the desired settings as a fresh settings file for first user restart.
